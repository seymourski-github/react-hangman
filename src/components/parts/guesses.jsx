import React from 'react';
import PropTypes from 'prop-types';
import { Button, Container, Input, Label } from '../../components';

export const Guesses = props => {

  const {
    action, id, labels, guessing, handleAction, handleInput
  } = props;

  const mapProps = name => ({
    name, handleAction,
    handleChange: handleInput,
    text: labels[name],
    style:  { flex: 'none' },
    value: guessing
  });

  return (
    <Container flex="row" id={id}>
      <Label alignSelf="flex-end" flexItem="100%" htmlFor="guessing" text={labels.guessing} />
      <Input {...mapProps('guessing', props)} autoFocus flexItem="1 0 auto" />
      <Button {...mapProps('guessed', props)} action={action} flexItem="none" />
    </Container>
  );

};

Guesses.propTypes = {
  action: PropTypes.string.isRequired,
  guessing: PropTypes.string,
  guessed: PropTypes.string,
  id: PropTypes.string.isRequired,
  labels: PropTypes.object,
  handleAction: PropTypes.func,
  handleInput: PropTypes.func
};

Guesses.defaultProps = {
  action: 'ACTION_OF_BUTTON',
  guessing: 'guess being typed and sent via handleInput',
  guessed: 'last guess sent using handleAction',
  labels: {
    input: 'Make a guess',
    button: 'ok'
  },
  handleAction: () => console.log('handleAction button fail'),
  handleInput: () => console.log('handleInput button fail'),
};
