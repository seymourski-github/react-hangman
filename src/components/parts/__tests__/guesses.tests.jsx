import React from 'react';
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { Guesses } from '../guesses';

const props = {
  id: 'my-guesses-panel',
  label: 'My guesses label'
};

describe('<Guesses /> Component', function() {

  const component = renderer.create(<Guesses {...props} />);
  let wrapper = shallow(<Guesses {...props} />);

  it("renders and matches our snapshot", () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
