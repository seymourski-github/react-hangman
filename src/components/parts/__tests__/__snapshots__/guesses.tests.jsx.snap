// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Guesses /> Component renders and matches our snapshot 1`] = `
<div
  className="container row"
  id="my-guesses-panel"
  style={undefined}
>
  <label
    className="label"
    disabled={false}
    hidden={false}
    htmlFor="guessing"
    style={
      Object {
        "alignSelf": "flex-end",
        "flex": "100%",
        "visibility": "visible",
      }
    }
  />
  <input
    className="input"
    disabled={false}
    id="guessing"
    name="guessing"
    onBlur={[Function]}
    onChange={[Function]}
    style={
      Object {
        "flex": "1 0 auto",
      }
    }
    type="text"
    value="guess being typed and sent via handleInput"
  />
  <button
    className="button"
    disabled={false}
    name="guessed"
    onClick={[Function]}
    style={null}
    type="button"
    value="guess being typed and sent via handleInput"
  >
    guessed
  </button>
</div>
`;
